src/components/molecules/OnlineStats/OnlineStats.tsx

```diff
@@ -78,6 +84,17 @@ const OnlineStats: React.FC = () => {
   const { recentVenueUsers } = useRecentVenueUsers();
 
   const venueName = venue?.name;
+  const dispatch = useDispatch();
+  const isUserProfileModalVisible = useSelector(
+    userProfileModalVisibilitySelector
+  );
+
+  const setUserProfileModalVisible = useCallback(() => {
+    if (isUserProfileModalVisible) {
+      setSelectedUserProfile(undefined);
+    }
+    dispatch(setUserProfileModalVisibility(!isUserProfileModalVisible));
+  }, [dispatch, isUserProfileModalVisible]);
```

Comment:
I would see this moved into `useProfileModalHook`, which would be quite similar to useChatSidebarControls. It would return the state, turn off/ turn on functions/ maybe a toggler?

Reply:
As per my comments in https://github.com/sparkletown/sparkle/pull/1218#pullrequestreview-597998714 most of this is likely not required. I spelled out the custom hook/etc stuff as well.